File: Text/check_palindrome/check_palindrome.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let input = input.trim();

    if input
        .chars()
        .zip(input.chars().rev())
        .all(|(c1, c2)| c1 == c2)
    {
        println!("That is a palindrome!");
    } else {
        println!("That is not a palindrome.");
    }
}

File: Text/check_palindrome/check_palindrome.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Please enter the string you want checked...")
  var input string
  fmt.Scan(&input)
  var rev string
  for i := len(input) - 1; i >= 0; i-- {
    rev += string(input[i])
  }
  if input == rev {
    fmt.Printf("%v is a palindrome\n", input)
  } else {
    fmt.Printf("%v is not a palindrome\n", input)
  }
}
